src/main/java/de/dennisguse/opentracks/services/GpsStatus.java

```diff
@@ -0,0 +1,217 @@
+package de.dennisguse.opentracks.services;
+
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationManager;
+import android.os.Handler;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.util.PreferencesUtils;
+
+/**
+ * This class handle GPS status according to received locations and some thresholds.
+ */
+class GpsStatus {
+
+    private static final String TAG = GpsStatus.class.getSimpleName();
+
+    // If recording interval not set in preferences then it uses 10 seconds like signal lost threshold.
+    private static final int SIGNAL_LOST_THRESHOLD_DEFAULT = 10000;
```

Comment:
is there already such a preference?

Reply:
In the service uses 30 seconds when user choose "Adapt for battery life" and "Adapt for accuracy":
`30 * UnitConversions.ONE_SECOND_MS`

But I think if GPS didn't receive any point for 10 seconds we can say that signal was lost. Because of that I created a constant in GpsStatus for the case the user don't select any min interval.